components/camel-rabbitmq/src/main/java/org/apache/camel/component/rabbitmq/RabbitConsumer.java

```diff
@@ -326,9 +340,13 @@ public void reconnect() throws Exception {
         } else if (channel == null || !isAutomaticRecoveryEnabled()) {
             LOG.info("Attempting to open a new rabbitMQ channel");
             Connection conn = consumer.getConnection();
-            channel = openChannel(conn);
-            // Register the channel to the tag
-            start();
+            try {
+                stop();
```

Comment:
Just wonder what to do if stop throw an exception, should we silently ignore this?

Reply:
The case I had during my integration test was that in the stop(), there is a call to "basicCancel", which throws an unhandled exception if the channel or the connection was already closed, which lead to "start()" never called, and thus the consumers never handled new message afterward.

Maybe the exception handling could be more specific to the basicCancel and not on all the "stop()", but I thought that start should always be called no matter how stop finishes. 

I'm open to other way of handling that.

Reply:
Yeah that sounds okay lets go with this - its better than whats before